windows/src/developer/TIKE/compile/CompileKeymanWeb.pas

```diff
@@ -909,24 +1028,65 @@ function TCompileKeymanWeb.JavaScript_ShiftAsString(fkp: PFILE_KEY; FMnemonic: B
 function TCompileKeymanWeb.JavaScript_Store(pwsz: PWideChar): string;
 var
   ch: DWord;
+  n: Integer;
+  rec: TSentinelRecord;
+const
+  wcsentinel: WideString = #$FFFF;
 begin
-  Result := '"';
-	while pwsz^ <> #0 do
+  n := Pos(wcsentinel, pwsz);
+
+  // Start:  plain text store.  Always use for 9.0-, conditionally for 10.0+.
+  if (n = 0) or not IsKeyboardVersion10OrLater then
   begin
-		if PWord(pwsz)^ = UC_SENTINEL then
+    Result := '"';
+    while pwsz^ <> #0 do
     begin
-      Result := Result + '.'; {TODO: fill in .}
-    end
-    else
-    begin
-      ch := GetSuppChar(pwsz);
-      if ch in [Ord('"'), Ord('\')] then Result := Result + '\';
-      Result := Result + Javascript_String(ch);  // I2242
+      if PWord(pwsz)^ = UC_SENTINEL then
+      begin
+        Result := Result + '.'; {TODO: fill in .}
+      end
+      else
+      begin
+        ch := GetSuppChar(pwsz);
+        if ch in [Ord('"'), Ord('\')] then Result := Result + '\';
+        Result := Result + Javascript_String(ch);  // I2242
+      end;
+
+      pwsz := incxstr(pwsz);
     end;
+    Result := Result + '"';
+  end
+  else
+  begin
+    Result := '[';
+    while pwsz^ <> #0 do
+    begin
+      if Result <> '[' then Result := Result + ',';
 
-    pwsz := incxstr(pwsz);
+      rec := ExpandSentinel(pwsz);
+      if rec.IsSentinel then
+      begin
+        if rec.Code = CODE_DEADKEY then
+        begin
+          Result := Result + Format('{d:%d}', [rec.Deadkey.DeadKey]);
+        end
+        else
+        begin
+          // How to report this error?  We should never see this.
```

Comment:
We have an error code for unexpected in global/inc/comperr.h:
````c++
#define CERR_SomewhereIGotItWrong		0x00008009
````

You'll need to add the definition to global/delphi/general/CompileErrorCodes.pas, then use it viz:

````pascal
ReportError(fkp.Line, CERR_SomewhereIGotItWrong...
````